File: algorithm/length_of_last_word/length_of_last_word.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-


class Solution(object):
    def lengthOfLastWord(self, s):
        """
        :type s: str
        :rtype: int
        """
        try:
            return len(s.split()[-1])
        except IndexError:
            return 0


if __name__ == '__main__':
    s = Solution()
    print s.lengthOfLastWord("Hello world")
    print s.lengthOfLastWord("a ")
    print s.lengthOfLastWord("b   a    ")

File: algorithm/length_of_last_word/length_of_last_word.go
package main

import (
	"fmt"
	"strings"
)

func lengthOfLastWord(s string) int {
	if len(s) == 0 {
		return 0
	}
	words := strings.Split(s, " ")
	pos := len(words) - 1
	for pos >= 0 && len(words[pos]) == 0 {
		pos -= 1
	}
	if pos < 0 {
		return 0
	}
	return len(words[pos])
}

func main() {
	testCases := []string{
		"Hello world",
		"",
		"a ",
		"b   a    ",
		" ",
	}
	for _, c := range testCases {
		fmt.Printf("%s last word length is %d\n", c, lengthOfLastWord(c))
	}

}
